device.go

```diff
@@ -90,10 +103,35 @@ func (d *device) InstallUpdate(image io.ReadCloser, size int64) error {
 		return syscall.ENOSPC
 	}
 
-	w, err := io.Copy(b, image)
+	ssz, err := b.SectorSize()
 	if err != nil {
-		log.Errorf("failed to write image data to device %v: %v",
+		log.Errorf("failed to read sector size of block device %s: %v",
 			inactivePartition, err)
+		return err
+	}
+
+	buf := make([]byte, ssz)
```

Comment:
I did take a look at `io.CopyN` initially but I believe it has the same limitation as `io.Copy`, and that is that it copies 32kb as maximum on each iteration, even if size argument is larger.